### PR TITLE
Feat/mod thread support

### PR DESCRIPTION
**Pull request by oed**
This was simple enough.
We need to deal with ethAdd -> subDID separately once we have updated the pinning strategy for them.

### CODE REVIEW

File: src/cacheService.js [hunk 1]
```diff
@@ -64,12 +64,14 @@ class CacheService {
   async getThread (req, res, next) {
     const spaceName = req.query.space
```
Comment (zachferland) [inline src/cacheService.js]: want to also allow API to get thread with full address? and offer both full address or space/name/mod/members

File: src/pinning.js [hunk 2]
```diff
@@ -175,16 +183,24 @@ class Pinning {
...
+    if (!address) {
+      address = (await this.orbitdb._determineAddress(name, 'feed', {
+        accessController: {
+          type: 'thread-access',
+          threadName: name,
+          members,
+          rootMod
+        }
+      }, false)).toString()
+    }
```
Comment (zachferland) [inline src/pinning.js]: 👍

File: src/cacheService.js (hunk 1 above)
Comment (oed) [reply src/cacheService.js]: Oh, didn't think about that. I can add it yeah!

**Comment by oed** [comment]
@zachferland I pushed an update. This also helped me find a bug where the threads cache would not work. Please have another look and make sure that all is alright!

File: src/__tests__/pinning.test.js [hunk 3]
```diff
@@ -231,6 +239,9 @@ describe('Pinning', () => {
       testClient.announceThread()
       await responsesPromise
       await dbSyncPromise
+      // for some reason there is an issue with the db not getting fully
+      // replicated in time even after the dbSyncPromise. Wait for 0.5 s
+      await new Promise((resolve, reject) => { setTimeout(resolve, 500) })
```
Comment (zachferland) [inline src/__tests__/pinning.test.js]: hmm, only in tests, or think this would show up as an issues in production?

**Comment by zachferland** [comment]
yeah looks good, and better organized / more readable now, not running it right now, but will get to test a lot on the deployed development instance

File: src/__tests__/pinning.test.js (hunk 3 above)
Comment (oed) [reply src/__tests__/pinning.test.js]: This has to do with the code in the `testClient` afaik.

**Pull request merged**